### DISCUSSION TITLE
Admin token change - expected behaviour?

### DISCUSSION THREAD

**Discussion by boomam** [dani-garcia/vaultwarden#3698] [Q&A] [answered]
Hi,  
On v1.28, is it expected that a change in a docker-compose config for the admin token/hash, does *not* update an existing token?  
For it to change for me, verifying via the config file, i had to change it in the admin GUI instead.

Is this expected behaviour?

Thanks!

--
Similar discussions [here](https://github.com/dani-garcia/vaultwarden/discussions/3454) and [here](https://github.com/dani-garcia/vaultwarden/discussions/3458), but they don't answer this one specific query that I can see.

**Comment by stefan0xC** [comment, marked as answer]
> Is this expected behaviour?

Yeah, `data/config.json` takes precedence. See https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview
